spec/trigger_lock_spec.rb

```diff
@@ -76,5 +76,11 @@
       expect(trigger_lock_2.locked?).to be(true)
       expect(trigger_lock_1.unlock).to be(false)
     end
+
+    it "should not acquire the lock if there is an error" do
```

Comment:
I guess you could raise on `ConnectionPool#with` but I'm not sure that's really better... what's wrong with raising on `refresh` ?